### PR TITLE
Adding an Inline Wrapper component

### PR DESCRIPTION
**Pull request by k-vosswinkel**
Adding "Inline Wrapper" (ArticleCallout in qz-react) for future use across the site.
Screen Shot 2020-08-05 at 4 21 39 PM
Screen Shot 2020-08-05 at 4 21 32 PM
Screen Shot 2020-08-05 at 4 21 23 PM

### CODE REVIEW

File: src/components/InlineWrapper/InlineWrapper.docs.mdx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Story, Preview } from '@storybook/addon-docs/blocks';
+import InlineWrapper from './InlineWrapper';
+
+# InlineWrapper
```
Comment (jshakes) [inline src/components/InlineWrapper/InlineWrapper.docs.mdx:4]: This name feels a bit vague and doesn't really tell me when it's appropriate to use this. Could we keep the Callout name? Shopify Polaris call theirs a CalloutCard which I kind of like. https://polaris.shopify.com/components/structure/callout-card

How about CalloutWrapper?

File: src/components/InlineWrapper/InlineWrapper.scss [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	padding: 24px 12px;
+
+	@include for-tablet-portrait-up {
+		margin: 40px 0;
+		padding: 28px;
+	}
+
+	@media print {
+		display: none;
+	}
```
Comment (jshakes) [inline src/components/InlineWrapper/InlineWrapper.scss:21]:
```suggestion
```
I think this is right in some cases, e.g. article callouts, but probably won't be universally applicable

File: src/components/InlineWrapper/InlineWrapper.docs.mdx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+## Usage
+
+```js
+import { InlineWrapper } from '@quartz/interface/src/components';
+```
+
+## Props
+
+This component accepts children as props and wraps them with a stylized container.
```
Comment (jshakes) [inline src/components/InlineWrapper/InlineWrapper.docs.mdx:16]:
```suggestion
<Props of={InlineWrapper} />
```
Use the Storybook Props component to generate a table of props (even thought there is only one in this case)

Comment (jshakes) [reply src/components/InlineWrapper/InlineWrapper.docs.mdx:16]: You can add a description of the prop in the component's propTypes, see the Button component for an example

File: src/components/InlineWrapper/InlineWrapper.scss [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+@import '~@quartz/styles/scss/media-queries';
+@import '~@quartz/styles/scss/fonts';
+
+.container {
+	@include font-maison-500-3;
+
+	background-color: $color-background-2;
+	border: $border-solid-decorative;
+	border-radius: $size-border-radius;
+	margin: 20px 0 40px 0;
```
Comment (jshakes) [inline src/components/InlineWrapper/InlineWrapper.scss:11]:
```suggestion
```
Let's keep the outer spacing the responsibility of the consumer (qz-react now has some handy SCSS tokens for this)

File: src/components/InlineWrapper/InlineWrapper.story.jsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+const styles = {
+	width: 400,
+	margin: '0 auto',
+};
+
+export const Default = () => (
+	<div style={styles}>
+		<InlineWrapper>
+			<p>An <strong>inline wrapper</strong> encapsulates components that add context or additional functionality to a page.</p>
```
Comment (jshakes) [inline src/components/InlineWrapper/InlineWrapper.story.jsx:21]: Could you use some 'real' content here that helps others understand when to use this component? Another example or two would be helpful too

File: src/components/InlineWrapper/InlineWrapper.docs.mdx (hunk 1 above)
Comment (k-vosswinkel) [reply src/components/InlineWrapper/InlineWrapper.docs.mdx:4]: I agree - I like callout card a lot. Wrapper is so vague - we wrap everything 😂

File: src/components/InlineWrapper/InlineWrapper.scss (hunk 2 above)
Comment (k-vosswinkel) [reply src/components/InlineWrapper/InlineWrapper.scss:21]: 👍

File: src/components/InlineWrapper/InlineWrapper.story.jsx (hunk 5 above)
Comment (k-vosswinkel) [reply src/components/InlineWrapper/InlineWrapper.story.jsx]: I added some more realistic content in. What did you have in mind re: other examples? `ArticleCallout` shows dark mode as an example, and we use `ArticleCallout` with forms, but we don't have the `ColorScheme` component or any of our form components in this repo.

**Comment by k-vosswinkel** [comment]
Hey @jshakes thanks for the feedback! I'm getting a failing accessibility test now, about the background contrast. Not quite sure how to address because we're using our color scheme. 
Screen Shot 2020-08-06 at 4 59 39 PM

File: src/components/CalloutCard/CalloutCard.jsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+		{children}
+	</div>
+);
+
+CalloutCard.propTypes = {
+	/**
+	 * Child nodes to be rendered as the inner content of the callout card.
+	 */
+	children: PropTypes.oneOfType( [
+		PropTypes.arrayOf( PropTypes.node ),
```
Comment (jshakes) [inline src/components/CalloutCard/CalloutCard.jsx:16]: Interesting, do you get a proptype error without this?

File: src/components/CalloutCard/CalloutCard.scss [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+.container {
+	@include font-maison-500-3;
+
+	background-color: $color-background-2;
+	border: $border-solid-decorative;
+	border-radius: $size-border-radius;
+	padding: 24px 12px;
+
+	@include for-tablet-portrait-up {
+		margin: 40px 0;
```
Comment (jshakes) [inline src/components/CalloutCard/CalloutCard.scss:14]:
```suggestion
```

File: src/components/CalloutCard/CalloutCard.scss [hunk 8]
```diff
@@ -0,0 +1,17 @@
+@import '~@quartz/styles/scss/borders';
```
Comment (jshakes) [inline src/components/CalloutCard/CalloutCard.scss:1]:
```suggestion
@import '~@quartz/styles/scss/borders';
@import '~@quartz/styles/scss/color-scheme';
```

File: src/components/CalloutCard/CalloutCard.story.jsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+	padding: '5px',
+};
+
+export const Form = () => (
+	<div style={styles}>
+		<CalloutCard>
+			<h3>Need to Know: Coronavirus</h3>
+			<p>Want a calm, rational, even curious approach to coronavirus? <a href="https://qz.com/emails/coronavirus/" target="_blank">We’ve got an email for that.</a> Sign up here:</p>
+			<label>Email address:
+				<input style={inputStyles} label="email" name="email" placeholder="email@email.com" />
```
Comment (jshakes) [inline src/components/CalloutCard/CalloutCard.story.jsx:41]: Could you add a todo comment to replace this with the real text input component once it's migrated? I will make a cooldown ticket for this

File: README.md [hunk 10]
```diff
@@ -57,6 +57,15 @@ If you want to **add a new component**:
...
 If the answer to any of these questions is no, chat with your teammates about how best to achieve what you want. Don’t be afraid to ask questions in the `#product-prism-support` Slack channel.
 
+### File Structure
+Each component should have a few files associated with it:
+- The component itself (`.jsx`)
+- Any necessary styling (`.scss`)
+- Component documentation (`.docs.mdx`)
+- Component stories (`.story.jsx`)
+
+The component and styles work together as you would expect in any React application. The documentation file comes from the Storybook Docs addon [(docs)](https://github.com/storybookjs/storybook/tree/master/addons/docs) and works directly with your stories [(docs)](https://github.com/storybookjs/storybook/tree/master).
```
Comment (jshakes) [inline README.md:67]: I'm not sure you need this imo, the list of files is probably enough

File: README.md [hunk 11]
```diff
@@ -57,6 +57,15 @@ If you want to **add a new component**:
 
 If the answer to any of these questions is no, chat with your teammates about how best to achieve what you want. Don’t be afraid to ask questions in the `#product-prism-support` Slack channel.
 
+### File Structure
+Each component should have a few files associated with it:
+- The component itself (`.jsx`)
+- Any necessary styling (`.scss`)
+- Component documentation (`.docs.mdx`)
+- Component stories (`.story.jsx`)
+
```
Comment (jshakes) [inline README.md:66]: Optionally a test file (`test.js`)

File: src/components/CalloutCard/CalloutCard.jsx (hunk 6 above)
Comment (k-vosswinkel) [reply src/components/CalloutCard/CalloutCard.jsx:16]: Oh I guess not - I'll simplify to node.

File: README.md (hunk 10 above)
Comment (k-vosswinkel) [reply README.md:67]: I wanted to explain addons somewhere. I'll just add a link to the list item

**Pull request merged**